src/cljdoc/doc_tree.clj

```diff
@@ -83,35 +95,36 @@
 (defmethod filepath->type "md" [_] :cljdoc/markdown)
 (defmethod filepath->type "adoc" [_] :cljdoc/asciidoc)
 
-(defn- process-toc-entry
-  [{:keys [slurp-fn get-contributors] :as fns}
-   {:keys [title attrs children]}]
-  {:pre [(string? title) (fn? slurp-fn) (fn? get-contributors)]}
+(defn- process-file-attrs [{:keys [slurp-fn get-contributors]}
+                           title {:keys [file slug]}]
```

Comment:
From a formatting perspective this might be easier readable as 
```suggestion
(defn- process-file-attrs 
  [{:keys [slurp-fn get-contributors]}
   title
   {:keys [file slug]}]
```
Generally when I have to add a linebreak between arguments I add one between every argument for consistency (although you'll probably find exceptions within this code base where I didn't do that :))